controller.go

```diff
@@ -547,6 +563,11 @@ func (c *controller) Sandboxes() []Sandbox {
 
 	list := make([]Sandbox, 0, len(c.sandboxes))
 	for _, s := range c.sandboxes {
+		// Hide stub sandboxes from libnetwork users
+		if s.isStub {
+			continue
+		}
```

Comment:
sandbox cleanup was happening properly until we added the retain logic via #724 which will retain the sandbox for host network case because for host and none the networks are not persisted so the sandbox gets retained(because the cleanup code fails to get the network since it won't be there) . That is the cause of the latest issue in docker/docker#17577